codegen/src/main/java/software/amazon/awssdk/codegen/poet/auth/scheme/AuthSchemeSpecUtils.java

```diff
@@ -82,6 +99,13 @@ public boolean usesSigV4() {
         return AuthUtils.usesAwsAuth(intermediateModel);
     }
 
+    public boolean useEndpointBasedAuthProvider() {
+        // Endpoint based auth provider is gated using the same setting that enables the use of auth scheme params. One does
+        // not make sense without the other so there's no much point on creating another setting if both have to be at the same
+        // time enabled or disabled.
+        return intermediateModel.getCustomizationConfig().isEnableEndpointAuthSchemeParams();
```

Comment:
should it `return generateEndpointBasedParams()`?

Reply:
It could, not sure why it should, changed anyways.